eva-accession-import-dbsnp2/src/main/java/uk/ac/ebi/eva/accession/dbsnp2/model/DbsnpJson.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2019 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.ac.ebi.eva.accession.dbsnp2.model;
+
+import org.springframework.batch.core.configuration.annotation.StepScope;
+import uk.ac.ebi.eva.accession.core.ISubmittedVariant;
+
+import java.time.LocalDateTime;
+
+/**
+ * Core fields of a variant extracted from a dbSNP JSON file.
+ */
+@StepScope
+public class DbsnpJson implements ISubmittedVariant {
```

Comment:
This should also extend AccessionedDocument<ISubmittedVariant, Long> to account for the SubSNP accession itself (which should also be captured btw. see **primary_snapshot_data->support** attribute with **id->type** as **subsnp**)

Reply:
I don't see the need for this class at all. Input objects should be converted to [SubmittedVariant](https://github.com/EBIvariation/eva-accession/blob/master/eva-accession-core/src/main/java/uk/ac/ebi/eva/accession/core/SubmittedVariant.java), which means that one can simply call the appropriate constructor from the processor.

For serialization, a [DbsnpVariantsWrapper](https://github.com/EBIvariation/eva-accession/blob/master/eva-accession-import/src/main/java/uk/ac/ebi/eva/accession/dbsnp/persistence/DbsnpVariantsWrapper.java) grouping both the submitted and clustered variant **database entities** is constructed and used as input for database serialization.

Please check the module `eva-accession-import` for reference on how the pipeline currently works, on we can go through it during one of the calls.